firefox-ios/Client/Frontend/Browser/Tabs/Views/CustomSelectorView/TabTraySelectorView.swift

```diff
@@ -70,71 +51,54 @@ final class TabTraySelectorView: UIView,
     }
 
     private func setup() {
-        addSubview(collectionView)
-        collectionView.translatesAutoresizingMaskIntoConstraints = false
+        let stackView = UIStackView()
+        stackView.axis = .horizontal
+        stackView.spacing = TabTraySelectorUX.horizontalPadding
+        stackView.distribution = .equalCentering
+        stackView.alignment = .center
+
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(stackView)
 
         NSLayoutConstraint.activate([
-            collectionView.topAnchor.constraint(equalTo: topAnchor, constant: -TabTraySelectorUX.verticalInsets),
-            collectionView.bottomAnchor.constraint(equalTo: bottomAnchor, constant: TabTraySelectorUX.verticalInsets),
-            collectionView.leadingAnchor.constraint(equalTo: leadingAnchor),
-            collectionView.trailingAnchor.constraint(equalTo: trailingAnchor)
+            stackView.centerYAnchor.constraint(equalTo: centerYAnchor),
+            stackView.centerXAnchor.constraint(equalTo: centerXAnchor)
         ])
 
-        applyTheme(theme: theme)
-    }
-
-    override func layoutSubviews() {
-        super.layoutSubviews()
+        for (index, title) in items.enumerated() {
+            let button = UIButton()
+            button.setTitle(title, for: .normal)
+            button.tag = index
+            button.addTarget(self, action: #selector(sectionSelected(_:)), for: .touchUpInside)
 
-        let estimatedCellWidth = TabTraySelectorUX.estimatedCellWidth
-        let horizontalInset = (bounds.width - estimatedCellWidth) / 2
+            buttons.append(button)
+            stackView.addArrangedSubview(button)
 
-        collectionView.contentInset = UIEdgeInsets(
-            top: 0,
-            left: horizontalInset,
-            bottom: 0,
-            right: horizontalInset
-        )
-        collectionView.contentOffset.x = -horizontalInset
-    }
+            button.titleLabel?.font = index == selectedIndex ?
+                FXFontStyles.Bold.body.scaledFont() :
+                FXFontStyles.Regular.body.scaledFont()
```

Comment:
I am not sure what we'll want to here for accessibility. The items are not scrollable anymore and cannot be read entirely when dynamic type is enabled. So this is kind of unusable if users have bigger font sizes. Can we think of a short term thing to make this usuable without too much complexity?

![Simulator Screenshot - iPhone 16 - 2025-04-29 at 19 27 31](https://github.com/user-attachments/assets/78b517b0-706c-422b-80a4-bc45811af02a)

Reply:
Created https://mozilla-hub.atlassian.net/browse/FXIOS-12187